Sources/PointFree/SiteMiddleware.swift

```diff
@@ -13,35 +14,69 @@ public let siteMiddleware: Middleware<StatusLineOpen, ResponseEnded, Prelude.Uni
       realm: "Point-Free",
       protect: isProtected
     )
+    <<< handleErrors
     <| render(conn:)
 
-private func render(conn: Conn<StatusLineOpen, Route>) -> IO<Conn<ResponseEnded, Data>> {
+private func render(conn: Conn<StatusLineOpen, Route>)
+  -> IO<Conn<ResponseEnded, Either<Error, Data>>> {
 
-  switch conn.data {
-  case let .githubCallback(code, redirect):
-    return conn.map(const((code, redirect)))
-      |> githubCallbackResponse
+    switch conn.data {
+    case let .githubCallback(code, redirect):
+      return conn.map(const((code, redirect)))
+        |> githubCallbackResponse
 
-  case let .home(signedUpSuccessfully):
-    return conn.map(const(signedUpSuccessfully))
-      |> homeResponse
+    case let .home(signedUpSuccessfully):
+      return conn.map(const(signedUpSuccessfully))
+        |> homeResponse
+        >>> map(map(pure))
```

Comment:
Cool to be able to look at our main routing function and see which endpoints are pure and which may have errors!